src/libImaging/GifDecode.c

```diff
@@ -264,29 +264,33 @@ ImagingGifDecode(Imaging im, ImagingCodecState state, UINT8* buffer, int bytes)
 	/* To squeeze some extra pixels out of this loop, we test for
 	   some common cases and handle them separately. */
 
-	/* FIXME: should we handle the transparency index in here??? */
-
-	if (i == 1) {
-	    if (state->x < state->xsize-1) {
-		/* Single pixel, not at the end of the line. */
-		*out++ = p[0];
-		state->x++;
+	/* If we have transparency, we need to use the regular loop. */
+	if(context->transparency == -1)
+	{
+	    if (i == 1) {
+		if (state->x < state->xsize-1) {
+		    /* Single pixel, not at the end of the line. */
+		    *out++ = p[0];
+		    state->x++;
+		    continue;
+		}
+	    } else if (state->x + i <= state->xsize) {
```

Comment:
Please, always use 4 spaces as indention everywhere in added code

Reply:
I'm using the same indentation style as the rest of the file: 8-space tabs with 4-space blocks (vim: noet sw=4 ts=8).